casadi: init at 3.6.6

nim65s opened the pull request:
## Description of changes

Seems to build and import fine, but this require additional testing, and checking logs todos

ref. https://github.com/NixOS/nixpkgs/pull/326696

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

nim65s committed:
proxsuite: init at 0.6.6

nim65s committed:
casadi: init at 3.6.5

nim65s committed:
superscs: init at 1.3.2

nim65s committed:
sleqp: init at 1.0.2

nim65s committed:
trlib: init at 0.4

nim65s committed:
osqp: fix cmake export

ref. https://github.com/osqp/osqp/pull/481
closes https://github.com/NixOS/nixpkgs/pull/235850

nim65s committed:
blasfeo: init at 0.1.3

nim65s committed:
hpipm: init at 0.1.3

nim65s committed:
fatrop: init at 0.0.1

nim65s committed:
mumps: init at 5.7.3

nim65s committed:
scotch: also install libesmumps for mumps

while here, update packaging to CMake, and build on darwin

Sigmanificient commented:
Good luck with porting Casadi from source, I will rebase my pr when you got it merge, to only have `neurafoil` & `aerosandbox`

nim65s committed:
spral: init at 2024.05.08

nim65s committed:
ipopt: enable mumps & spral

nim65s committed:
pinocchio: add casadiSupport

nim65s commented:
My goal here is to get casadi working on pinocchio. It seems that is fine, at least on x86_64-linux with `example-py-casadi-quadrotor-ocp`  which use casadi + ipopt + mumps from python:
```
pinocchio> EXIT: Optimal Solution Found.
pinocchio>       solver  :   t_proc      (avg)   t_wall      (avg)    n_eval
pinocchio>        nlp_f  | 103.75ms (  6.92ms)   4.73ms (315.28us)        15
pinocchio>        nlp_g  |  78.18ms (  5.21ms)   8.84ms (589.47us)        15
pinocchio>   nlp_grad_f  | 216.33ms ( 14.42ms)  12.38ms (825.32us)        15
pinocchio>    nlp_jac_g  |   2.40 s (160.20ms) 148.40ms (  9.89ms)        15
pinocchio>        total  | 300.93 s (300.93 s)  15.22 s ( 15.22 s)         1
```

I'll check what darwin says now.

taktoa reviewed (commented):
Changes to osqp look fine

nim65s commented:
Result of `nixpkgs-review pr 331343` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>blasfeo</li>
    <li>fatrop</li>
    <li>hpipm</li>
    <li>osqp</li>
    <li>proxsuite</li>
    <li>proxsuite.doc</li>
    <li>python311Packages.fatrop</li>
    <li>python311Packages.proxsuite</li>
    <li>python311Packages.proxsuite.doc</li>
    <li>python311Packages.sleqp</li>
    <li>python311Packages.trlib</li>
    <li>python312Packages.fatrop</li>
    <li>python312Packages.proxsuite</li>
    <li>python312Packages.proxsuite.doc</li>
    <li>python312Packages.sleqp</li>
    <li>python312Packages.trlib</li>
    <li>sleqp</li>
    <li>spral</li>
    <li>superscs</li>
    <li>trlib</li>
  </ul>
</details>

nim65s commented:
Result of `nixpkgs-review pr 331343` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>19 packages failed to build:</summary>
  <ul>
    <li>bonmin</li>
    <li>casadi</li>
    <li>crocoddyl</li>
    <li>ipopt</li>
    <li>openturns</li>
    <li>pagmo2</li>
    <li>pinocchio</li>
    <li>python311Packages.casadi</li>
    <li>python311Packages.crocoddyl</li>
    <li>python311Packages.example-robot-data</li>
    <li>python311Packages.openturns</li>
    <li>python311Packages.pinocchio</li>
    <li>python311Packages.pygmo</li>
    <li>python312Packages.casadi</li>
    <li>python312Packages.crocoddyl</li>
    <li>python312Packages.example-robot-data</li>
    <li>python312Packages.openturns</li>
    <li>python312Packages.pinocchio</li>
    <li>python312Packages.pygmo</li>
  </ul>
</details>
<details>
  <summary>26 packages built:</summary>
  <ul>
    <li>blasfeo</li>
    <li>fatrop</li>
    <li>hpipm</li>
    <li>mumps</li>
    <li>osqp</li>
    <li>proxsuite</li>
    <li>proxsuite.doc</li>
    <li>python311Packages.fatrop</li>
    <li>python311Packages.fenics</li>
    <li>python311Packages.fenics.dist</li>
    <li>python311Packages.proxsuite</li>
    <li>python311Packages.proxsuite.doc</li>
    <li>python311Packages.sleqp</li>
    <li>python311Packages.trlib</li>
    <li>python312Packages.fatrop</li>
    <li>python312Packages.fenics</li>
    <li>python312Packages.fenics.dist</li>
    <li>python312Packages.proxsuite</li>
    <li>python312Packages.proxsuite.doc</li>
    <li>python312Packages.sleqp</li>
    <li>python312Packages.trlib</li>
    <li>scotch</li>
    <li>sleqp</li>
    <li>spral</li>
    <li>superscs</li>
    <li>trlib</li>
  </ul>
</details>

nim65s commented:
@Sigmanificient : this lgtm now. Maybe you could double check that what you need in your PR is fine with this ?

nim65s commented:
Ok, this just broke after https://github.com/NixOS/nixpkgs/pull/327438…

nim65s commented:
Result of `nixpkgs-review pr 331343` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>blasfeo</li>
    <li>fatrop</li>
    <li>hpipm</li>
    <li>osqp</li>
    <li>proxsuite</li>
    <li>proxsuite.doc</li>
    <li>python311Packages.fatrop</li>
    <li>python311Packages.proxsuite</li>
    <li>python311Packages.proxsuite.doc</li>
    <li>python311Packages.sleqp</li>
    <li>python311Packages.trlib</li>
    <li>python312Packages.fatrop</li>
    <li>python312Packages.proxsuite</li>
    <li>python312Packages.proxsuite.doc</li>
    <li>python312Packages.sleqp</li>
    <li>python312Packages.trlib</li>
    <li>sleqp</li>
    <li>spral</li>
    <li>superscs</li>
    <li>trlib</li>
  </ul>
</details>

wegank commented:
Result of `nixpkgs-review pr 331343` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>17 packages marked as broken and skipped:</summary>
  <ul>
    <li>bonmin</li>
    <li>crocoddyl</li>
    <li>ipopt</li>
    <li>openturns</li>
    <li>pinocchio</li>
    <li>python311Packages.crocoddyl</li>
    <li>python311Packages.example-robot-data</li>
    <li>python311Packages.fenics</li>
    <li>python311Packages.fenics.dist</li>
    <li>python311Packages.openturns</li>
    <li>python311Packages.pinocchio</li>
    <li>python312Packages.crocoddyl</li>
    <li>python312Packages.example-robot-data</li>
    <li>python312Packages.fenics</li>
    <li>python312Packages.fenics.dist</li>
    <li>python312Packages.openturns</li>
    <li>python312Packages.pinocchio</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>osqp</li>
  </ul>
</details>

wegank commented on pkgs/by-name/pr/proxsuite/package.nix:
```diff
@@ -0,0 +1,97 @@
+    # Allow use of system pybind11
+    # upstream will move to nanobind for next release, so this can be dismissed
+    (fetchpatch {
+      url = "https://github.com/Simple-Robotics/proxsuite/pull/337/commits/725d8bd31e584ed4dfdf75f78e0b8f46d5b4918f.patch";
```
404ed

nim65s replied:
Thanks, fixed.

nim65s commented:
I rebased to fix conflict following https://github.com/NixOS/nixpkgs/pull/325214, and also removed some darwin support. I thought it was working, but some openmpi issues were hidden, ref. https://github.com/NixOS/nixpkgs/pull/332983

nim65s commented:
Result of `nixpkgs-review pr 331343` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>bonmin</li>
  </ul>
</details>
<details>
  <summary>34 packages built:</summary>
  <ul>
    <li>blasfeo</li>
    <li>crocoddyl</li>
    <li>fatrop</li>
    <li>hpipm</li>
    <li>ipopt</li>
    <li>openturns</li>
    <li>osqp</li>
    <li>pinocchio</li>
    <li>proxsuite</li>
    <li>proxsuite.doc</li>
    <li>python311Packages.crocoddyl</li>
    <li>python311Packages.example-robot-data</li>
    <li>python311Packages.fatrop</li>
    <li>python311Packages.openturns</li>
    <li>python311Packages.pinocchio</li>
    <li>python311Packages.proxsuite</li>
    <li>python311Packages.proxsuite.doc</li>
    <li>python311Packages.pygmo</li>
    <li>python311Packages.sleqp</li>
    <li>python311Packages.trlib</li>
    <li>python312Packages.crocoddyl</li>
    <li>python312Packages.example-robot-data</li>
    <li>python312Packages.fatrop</li>
    <li>python312Packages.openturns</li>
    <li>python312Packages.pinocchio</li>
    <li>python312Packages.proxsuite</li>
    <li>python312Packages.proxsuite.doc</li>
    <li>python312Packages.pygmo</li>
    <li>python312Packages.sleqp</li>
    <li>python312Packages.trlib</li>
    <li>sleqp</li>
    <li>spral</li>
    <li>superscs</li>
    <li>trlib</li>
  </ul>
</details>

nim65s commented:
Result of `nixpkgs-review pr 331343` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>45 packages built:</summary>
  <ul>
    <li>blasfeo</li>
    <li>bonmin</li>
    <li>casadi</li>
    <li>crocoddyl</li>
    <li>fatrop</li>
    <li>hpipm</li>
    <li>ipopt</li>
    <li>mumps</li>
    <li>openturns</li>
    <li>osqp</li>
    <li>pagmo2</li>
    <li>pinocchio</li>
    <li>proxsuite</li>
    <li>proxsuite.doc</li>
    <li>python311Packages.casadi</li>
    <li>python311Packages.crocoddyl</li>
    <li>python311Packages.example-robot-data</li>
    <li>python311Packages.fatrop</li>
    <li>python311Packages.fenics</li>
    <li>python311Packages.fenics.dist</li>
    <li>python311Packages.openturns</li>
    <li>python311Packages.pinocchio</li>
    <li>python311Packages.proxsuite</li>
    <li>python311Packages.proxsuite.doc</li>
    <li>python311Packages.pygmo</li>
    <li>python311Packages.sleqp</li>
    <li>python311Packages.trlib</li>
    <li>python312Packages.casadi</li>
    <li>python312Packages.crocoddyl</li>
    <li>python312Packages.example-robot-data</li>
    <li>python312Packages.fatrop</li>
    <li>python312Packages.fenics</li>
    <li>python312Packages.fenics.dist</li>
    <li>python312Packages.openturns</li>
    <li>python312Packages.pinocchio</li>
    <li>python312Packages.proxsuite</li>
    <li>python312Packages.proxsuite.doc</li>
    <li>python312Packages.pygmo</li>
    <li>python312Packages.sleqp</li>
    <li>python312Packages.trlib</li>
    <li>scotch</li>
    <li>sleqp</li>
    <li>spral</li>
    <li>superscs</li>
    <li>trlib</li>
  </ul>
</details>
